sycl/source/detail/kernel_impl.cpp

```diff
@@ -93,6 +93,35 @@ bool kernel_impl::isCreatedFromSource() const {
   return MCreatedFromSource;
 }
 
+void kernel_impl::checkIfValidForNumArgsInfoQuery() const {
+  bool isBuiltIn = false;
+  std::string KernelName = get_info<info::kernel::function_name>();
+  if (KernelName != "") {
```

Comment:
```suggestion
  if (!KernelName.empty()) {
```

Reply:
I removed this check, because it looks like this check have no sense, since `KernelName` can't be empty.